packages/table/src/headers/_headers.scss

```diff
@@ -65,18 +78,35 @@
     mix($pt-intent-primary, $dark-header-hover-background-color, 10%)
   );
 
-  flex-shrink: 0;
   position: absolute;
   top: 0;
-  right: $cell-border-width; // don't overlap box-shadow border
-  opacity: 0; // hide until header is hovered
-  background: none;
+  right: 0;
+  width: $column-header-min-height + $pt-grid-size * 2;
+  height: $column-header-min-height;
+  pointer-events: none; // don't swallow clicks meant for the underlying header cell
+
+  .bp-table-header:hover &,
+  .bp-table-header-active & {
+    background-image: $menu-hover-background;
+
+    .pt-dark & {
+      background-image: $dark-menu-hover-background;
+    }
+  }
+
+  .bp-table-header.bp-table-header-selected & {
+    background-image: $menu-selected-background;
+
+    .pt-dark & {
+      background-image: $dark-menu-selected-background;
+    }
+  }
+}
+
+.bp-table-th-menu {
   cursor: $action-cursor;
-  width: $column-header-min-height + $pt-grid-size;
+  width: $column-header-min-height; // create a larger, more accessible click target
```

Comment:
i don't think comment means much, and will mean even less in a month. it doesn't even look correct when i compare with code on the left, where the width _was_ obviously larger.